src/FieldValidator.php

```diff
@@ -148,4 +149,63 @@ private function assertValidType(string $type, $value): bool
     {
         return $value instanceof $type || gettype($value) === $type;
     }
+
+    protected function resolveAllowedArrayTypesFromCollection(string $type): array
+    {
+        if (! $type || ! class_exists($type) || ! is_subclass_of($type, DataTransferObjectCollection::class)) {
+            return [];
+        }
+
+        $class = new ReflectionClass($type);
+
+        $currentReturnType = $class->getMethod('current')->getReturnType();
+
+        // We cast to array to support future union types in PHP 8
+        $currentReturnTypes = [];
+        if ($currentReturnType) {
+            $currentReturnTypes[] = $currentReturnType->getName();
+        }
+
+        $docblockReturnTypes = $class->getDocComment()
```

Comment:
I don't think I want to support the `/** @method … current */` notation on the class itself. That's extra overhead for something that's always possible by overriding the method itself no?

Reply:
Correct, and if we remove this then it will vastly simplify the PR, going to attempt this now